lib/parser-stats.js

```diff
@@ -0,0 +1,65 @@
+'use strict';
+
+const promisify = require('bluebird').promisify;
+const Table = require('cli-table');
+const glob = promisify(require('glob').glob);;
+const filesize = require('filesize');
+const percent = require('calc-percent');
+const fs = require('fs');
+
+
+var table = new Table({
+	head: ['files', 'original size', 'optimized size', 'saving'],
+});
+
+function stats(config) {
+	return Promise.all(
+			Object.keys(config.parsers).map(function(name){
+				return Promise.all([
+					getSumOfFileSize(config.src, config.parsers[name].pattern),
+					getSumOfFileSize(config.temp, config.parsers[name].pattern)
+				])
+			})
+		).then(function (sizes) {
```

Comment:
move function outside this pyramid, to `createSizesTable(sizes)`. I would also initiate the table (line 11) inside that function. Smaller, easier to test.